tower/src/make/make_service.rs

```diff
@@ -65,3 +147,51 @@ where
         Service::call(self, target)
     }
 }
+
+#[derive(Debug, Clone)]
```

Comment:
Unfortunately, I think we should probably provide manual `Clone` and `Debug` impls for this, rather than deriving them. The derived implementations require _all_ type parameters of the type deriving `Clone` or `Debug` to implement those traits. However, we don't actually need the request type to implement `Clone` to clone an `IntoService` or implement `Debug` to format one — it's only necessary because of the `PhantomData`.

Therefore, I think we should provide an `impl<M: Clone, Request> Clone for IntoService<M, Request>` and an `impl<M: fmt::Debug, Request> fmt::Debug for IntoService<M, Request>`, rather than deriving them.

Reply:
Ah yes ofc. Fixed in 4b1dd1505be29